File: backend/app/schemas/terms.py
from datetime import datetime

from pydantic import BaseModel


class TermsStatusRead(BaseModel):
    accepted: bool
    current_version: str
    accepted_version: str | None = None
    accepted_at: datetime | None = None


class AcceptTermsRequest(BaseModel):
    terms_version: str


class AcceptTermsResponse(BaseModel):
    accepted: bool
    terms_version: str
    accepted_at: datetime

File: frontend/src/types/terms.ts
export interface TermsStatus {
    accepted: boolean;
    current_version: string;
    accepted_version: string | null;
    accepted_at: string | null;
}

export interface AcceptTermsPayload {
    terms_version: string;
}

export interface AcceptTermsResponse {
    accepted: boolean;
    terms_version: string;
    accepted_at: string;
}
